src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -397,11 +397,12 @@ protected override void VisitSwitchSection(BoundSwitchSection node, bool isLastS
                                 if (_variableTypes.TryGetValue(local, out var existingType))
                                 {
                                     // merge inferred nullable annotation from different branches of the decision tree
-                                    _variableTypes[local] = TypeWithAnnotations.Create(existingType.Type, existingType.NullableAnnotation.Join(inferredType.NullableAnnotation));
+                                    inferredType = TypeWithAnnotations.Create(existingType.Type, existingType.NullableAnnotation.Join(inferredType.NullableAnnotation));
                                 }
-                                else
+
+                                if (local.IsVar)
```

Comment:
>if (local.IsVar) [](start = 32, length = 16)

Would it make sense to move all code that uses ```variableAccess``` local into this ```if```? It looks like executing it will not have any useful side-effects unless we get into this ```if```.

Reply:
This code is no longer changed from existing, will leave as is.

---
In reply to: [327847841](https://github.com/dotnet/roslyn/pull/38769#discussion_r327847841) [](ancestors = 327847841)